controllers/singlePageCrawler.js

```diff
@@ -25,14 +25,24 @@ async function scrapeSingleURL(productURL) {
       const description = $('div.product--description').text();
 
       console.log(title, description, img, price, numPrice);
-      const crawledProduct = new Product({
+      const updatedProduct = {
         url: productURL,
         title: title,
         description: description,
         img: img,
         price: numPrice
-      });
-      await crawledProduct.save();
+      };
+      console.log(updatedProduct);
+      const doc = await Product.findOne({ _id: id });
+      await Product.updateOne(
+        { _id: doc._id },
+        {
+          title: updatedProduct.title,
```

Comment:
Did you miss `url` here?

You could bypass the whole `updatedProduct`:

```
await Product.updateOne(
        { _id: id },
        updatedProduct
      );
```

Reply:
Thought `url` will stay the same on the object. Finding a product with a different `url` on initial run of the deep crawler will result in multiple products of the same type. Will handle this in future. Opened #73 for this issue.

This was my first intention. Ive made a mistake passing `updatedProducts` as an object.  My first attempt ->`await Product.updateOne({ _id: doc._id }, {updatedProduct});`
Solution -> `wait Product.updateOne({ _id: doc._id }, updatedProduct);`

Thx for advice!